--- xmms2-0.1DR2.1/src/clients/sdl-vis/Program.orig	2005-08-18 20:26:23.000000000 +0200
+++ xmms2-0.1DR2.1/src/clients/sdl-vis/Program	2023-05-06 21:45:00.043182408 +0200
@@ -9,3 +9,4 @@ def config(env):
 	env.pkgconfig("glib-2.0")
 	env.configcmd("sdl-config --libs --cflags")
 	env.checklib("SDL_ttf", "TTF_Init")
+	env.checklib("m", "log10")
--- xmms2-0.1DR2.1/src/clients/lib/xmmsclient-glib/Library.orig	2005-08-18 20:26:23.000000000 +0200
+++ xmms2-0.1DR2.1/src/clients/lib/xmmsclient-glib/Library	2023-05-07 22:35:37.261992755 +0200
@@ -5,3 +5,5 @@ source=["xmmsclient-glib.c"]
 
 def config(env):
 	env.pkgconfig("glib-2.0")
+	env.Prepend(LIBPATH="src/clients/lib/xmmsclient")
+	env.Append(LIBS=["xmmsclient"])
