--- xmms2-0.5DrLecter/src/xmms/main.c.orig	2008-06-15 19:31:38.000000000 +0200
+++ xmms2-0.5DrLecter/src/xmms/main.c	2023-05-21 17:23:28.557155999 +0200
@@ -420,7 +420,7 @@ main (int argc, char **argv)
 	}
 	if (showhelp) {
 #if GLIB_CHECK_VERSION(2,14,0)
-		g_print (g_option_context_get_help (context, TRUE, NULL));
+		g_print ("%s", g_option_context_get_help (context, TRUE, NULL));
 		exit (EXIT_SUCCESS);
 #else
 		g_print ("Please use --help or -? for help\n");
--- xmms2-0.6DrMattDestruction/src/clients/nycli/column_display.c.orig	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction/src/clients/nycli/column_display.c	2023-06-22 20:49:36.616801277 +0200
@@ -178,11 +178,11 @@ print_fixed_width_string (gchar *value,
                           column_def_align_t align, gchar padchar)
 {
 	if (align == COLUMN_DEF_ALIGN_LEFT) {
-		g_printf (value);
+		g_printf ("%s", value);
 		print_padding (width - realsize, padchar);
 	} else {
 		print_padding (width - realsize, padchar);
-		g_printf (value);
+		g_printf ("%s", value);
 	}
 }
 
@@ -200,7 +200,7 @@ print_string_using_coldef (column_displa
 
 	case COLUMN_DEF_SIZE_AUTO:
 		/* Just print the string */
-		g_printf (disp->buffer);
+		g_printf ("%s", disp->buffer);
 		break;
 	}
 }
@@ -360,7 +360,7 @@ column_display_print_header (column_disp
 	/* Display column headers */
 	for (i = 0; i < disp->cols->len; ++i) {
 		coldef = g_array_index (disp->cols, column_def_t *, i);
-		realsize = g_snprintf (disp->buffer, coldef->size + 1, coldef->name);
+		realsize = g_snprintf (disp->buffer, coldef->size + 1, "%s", coldef->name);
 		print_fixed_width_string (disp->buffer, coldef->size, realsize,
 		                          coldef->align, ' ');
 	}
@@ -524,7 +524,7 @@ column_display_render_text (column_displ
 	gint realsize;
 	const gchar *sep = coldef->name;
 
-	realsize = g_printf (sep);
+	realsize = g_printf ("%s", sep);
 
 	return realsize;
 }
@@ -545,7 +545,7 @@ column_display_render_time (column_displ
 		break;
 	default:
 		/* Invalid type, don't render anything*/
-		return;
+		return 0;
 	}
 
 	time = format_time (millisecs, FALSE);
@@ -557,7 +557,7 @@ column_display_render_time (column_displ
 	 * alignment and overflow termwidth if strlen(time) > coldef->size,
 	 * but it's a temporary fix to avoid displaying a wrong time.
 	 */
-	realsize = g_snprintf (disp->buffer, disp->termwidth + 1, time);
+	realsize = g_snprintf (disp->buffer, disp->termwidth + 1, "%s", time);
 	print_string_using_coldef (disp, coldef, realsize);
 
 	g_free (time);
--- xmms2-0.6DrMattDestruction/src/clients/lib/ruby/rb_xmmsclient.c.orig	2023-06-22 20:50:00.663337673 +0200
+++ xmms2-0.6DrMattDestruction/src/clients/lib/ruby/rb_xmmsclient.c	2023-06-22 20:53:19.725592593 +0200
@@ -166,7 +166,7 @@ c_connect (int argc, VALUE *argv, VALUE
 		snprintf (buf, sizeof (buf), "cannot connect to daemon (%s)\n",
 		          xmmsc_get_last_error (xmms->real));
 
-		rb_raise (eClientError, buf);
+		rb_raise (eClientError, "%s", buf);
 	}
 
 	return self;
