--- xmms2-0.2DrAlban/SConstruct.orig	2005-12-18 17:35:27.000000000 +0100
+++ xmms2-0.2DrAlban/SConstruct	2023-05-09 21:43:50.776017416 +0200
@@ -72,8 +72,8 @@ base_env.Append(CPPPATH=["#src/include"]
 Help(opts.GenerateHelpText(base_env))
 
 def do_subst_in_file(targetfile, sourcefile, dict):
-	"""Replace all instances of the keys of dict with their values.
-	For example, if dict is {'%VERSION%': '1.2345', '%BASE%': 'MyProg'},
+        """Replace all instances of the keys of dict with their values.
+        For example, if dict is {'%VERSION%': '1.2345', '%BASE%': 'MyProg'},
         then all instances of %VERSION% in the file will be replaced with 1.2345 etc.
         """
         try:
@@ -135,10 +135,10 @@ base_env.Depends('#src/xmms/sample.c', '
 base_env.SourceCode('src/xmms/converter.c', b)
 
 subst_dict = {"%VERSION%":XMMS_VERSION, "%PLATFORM%":"XMMS_OS_" + base_env.platform.upper(), 
-	      "%PKGLIBDIR%":base_env["PREFIX"]+"/lib/xmms2",
-	      "%BINDIR%":base_env["PREFIX"]+"/bin",
-	      "%SHAREDDIR%":base_env.sharepath,
-	      "%PREFIX%":base_env.install_prefix}
+              "%PKGLIBDIR%":base_env["PREFIX"]+"/lib/xmms2",
+              "%BINDIR%":base_env["PREFIX"]+"/bin",
+              "%SHAREDDIR%":base_env.sharepath,
+              "%PREFIX%":base_env.install_prefix}
 
 config = base_env.SubstInFile("src/include/xmms/xmms_defs.h", "src/include/xmms/xmms_defs.h.in", SUBST_DICT=subst_dict)
 
@@ -160,7 +160,7 @@ def scan_headers(name):
 		newf = dir+"/"+d
 		if os.path.isfile(newf) and newf.endswith('.h'):
 			base_env.add_header(name, newf)
-			
+
 scan_headers("xmmsc")
 scan_headers("xmms")
 scan_headers("xmmsclient")
@@ -185,9 +185,9 @@ base_env.add_manpage(8, 'doc/xmms2d.8')
 #### Generate pc files.
 
 pc_files = [{"name": "xmms2-plugin", "lib":""}, 
-	    {"name":"xmms2-client", "lib":"-lxmmsclient"},
-	    {"name":"xmms2-client-glib", "lib":"-lxmmsclient-glib"},
-	    {"name":"xmms2-client-ecore", "lib":"-lxmmsclient-ecore"}]
+            {"name":"xmms2-client", "lib":"-lxmmsclient"},
+            {"name":"xmms2-client-glib", "lib":"-lxmmsclient-glib"},
+            {"name":"xmms2-client-ecore", "lib":"-lxmmsclient-ecore"}]
 
 for p in pc_files:
 	d = subst_dict.copy()
--- xmms2-0.2DrAlban/xmmsenv.py.orig	2023-05-09 21:16:22.821611818 +0200
+++ xmms2-0.2DrAlban/xmmsenv.py	2023-05-09 21:45:23.205516682 +0200
@@ -352,7 +352,7 @@ class XMMSEnvironment(Environment):
 	def add_header(self, target, source):
 		self.Install(os.path.join(self.includepath,target), source)
 
-        def add_manpage(self, section, source):
+	def add_manpage(self, section, source):
 		self.GZipper(source + '.gz', source)
 		self.Install(os.path.join(self.manpath, "man"+str(section)), source+'.gz')
 
