--- xmms2-0.2DrAlban/src/plugins/sid/sid.c.orig	2005-12-18 17:35:27.000000000 +0100
+++ xmms2-0.2DrAlban/src/plugins/sid/sid.c	2023-05-10 17:17:13.484532781 +0200
@@ -192,10 +192,12 @@ xmms_sid_get_media_info (xmms_decoder_t
 
 	err = sidplay_wrapper_songinfo (data->wrapper, artist, title);
 	if (err > 0) {
-		xmms_medialib_entry_property_set_str (entry, 
+		xmms_medialib_session_t *session = xmms_medialib_begin ();
+		xmms_medialib_entry_property_set_str (session, entry, 
 		         XMMS_MEDIALIB_ENTRY_PROPERTY_TITLE, title);
-		xmms_medialib_entry_property_set_str (entry, 
+		xmms_medialib_entry_property_set_str (session, entry, 
 		         XMMS_MEDIALIB_ENTRY_PROPERTY_ARTIST, artist);
+		xmms_medialib_end (session);
 		xmms_medialib_entry_send_update (entry);
 	} 
 }
