--- xmms2-0.2DrDolittle/src/clients/lib/java/Library.orig	2023-05-11 12:50:52.113122775 +0200
+++ xmms2-0.2DrDolittle/src/clients/lib/java/Library	2023-05-11 20:55:45.265511648 +0200
@@ -103,7 +103,7 @@ def config(env):
 	
 	env.Java(target = mydir, source = [mydir + '/src'])
 	env.Jar(target=mydir + '/xmms2java.jar', source=mydir + '/org')
-	env.Install(env['PREFIX'] + 'share/xmms2/java', mydir + '/xmms2java.jar')
+	env.Install(env['PREFIX'] + '/share/java', mydir + '/xmms2java.jar')
 	def mkdir(target, source, env):
 		try:
 			os.mkdir(mydir + "/src/org/xmms2/xmms2bindings")
