--- xmms2-0.7DrNo/wscript.orig	2023-06-23 06:29:12.321345610 +0200
+++ xmms2-0.7DrNo/wscript	2023-06-23 06:29:55.905086544 +0200
@@ -273,7 +273,7 @@ def configure(conf):
         if changed:
             dirty="-dirty"
         conf.check_message("uncommitted changes", "", bool(changed))
-        conf.env["VERSION"] = BASEVERSION + " (git commit: %s%s)" % (nam, dirty)
+        conf.env["VERSION"] = BASEVERSION
 
     for warning in ('all',
                     'no-format-extra-args',
