--- xmms2-0.2DrGonzo/SConstruct.orig	2023-05-13 21:17:48.115540583 +0200
+++ xmms2-0.2DrGonzo/SConstruct	2023-05-13 21:25:55.299567947 +0200
@@ -181,8 +181,8 @@ scan_headers("xmmsclient/xmmsclient++")
 base_env.add_manpage(1, 'doc/xmms2.1')
 base_env.add_manpage(8, 'doc/xmms2d.8')
 base_env.add_manpage(1, 'doc/xmms2-et.1')
-base_env.add_manpage(1, 'doc/xmms2-launcher.8')
-base_env.add_manpage(1, 'doc/xmms2-mdns-avahi.8')
+base_env.add_manpage(8, 'doc/xmms2-launcher.8')
+base_env.add_manpage(8, 'doc/xmms2-mdns-avahi.8')
 
 #### Generate pc files.
 
