--- xmms2-0.2DrGonzo/src/clients/lib/java/Library.orig	2023-05-13 18:41:07.899799253 +0200
+++ xmms2-0.2DrGonzo/src/clients/lib/java/Library	2023-05-13 18:43:31.892352512 +0200
@@ -125,7 +125,7 @@ def config(env):
 	
 	env.Java(target = mydir, source = [mydir + '/src'])
 	env.Jar(target=mydir + '/xmms2java.jar', source=mydir + '/org')
-	env.Install(env['PREFIX'] + '/share/xmms2/java', mydir + '/xmms2java.jar')
+	env.Install(env['PREFIX'] + '/share/java', mydir + '/xmms2java.jar')
 	def mkdir(target, source, env):
 		try:
 			os.mkdir(mydir + "/src/org/xmms2/wrapper/xmms2bindings")
