--- xmms2-0.5DrLecter/src/clients/mdns/dns_sd/wscript.orig	2023-05-21 15:05:21.078719788 +0200
+++ xmms2-0.5DrLecter/src/clients/mdns/dns_sd/wscript	2023-05-21 15:06:42.958276208 +0200
@@ -17,7 +17,6 @@ def build(bld):
   obj = bld.create_obj('subst')
   obj.source = 'xmms2-mdns-launcher.sh.in'
   obj.dict = {'PRG_PATH' : env['BINDIR']}
-  Common.install_files('SCRIPTDIR', None, 'xmms2-mdns-launcher.sh', chmod=0755)
 
 def configure(conf):
   conf.env['explicit_dnssd'] = True
