--- xmms2-0.6DrMattDestruction/src/include/xmmsclient/xmmsclient++/signal.h.orig	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction/src/include/xmmsclient/xmmsclient++/signal.h	2023-06-22 18:59:11.362344508 +0200
@@ -18,7 +18,7 @@
 #define XMMSCLIENTPP_SIGNAL_H
 
 #include <xmmsclient/xmmsclient.h>
-#include <boost/signal.hpp>
+#include <boost/signals2.hpp>
 #include <string>
 #include <list>
 #include <iostream>
@@ -33,7 +33,7 @@ namespace Xmms
 		class Coll;
 	}
 
-	typedef boost::signal< bool( const std::string& ) > error_sig;
+	typedef boost::signals2::signal< bool( const std::string& ) > error_sig;
 
 	/** @class SignalInterface
 	 *  This is here only to unify all Signal classes so that they can be
@@ -54,7 +54,7 @@ namespace Xmms
 	template< typename T >
 	struct Signal : public SignalInterface
 	{
-		typedef boost::signal< bool( const T& ) > signal_t;
+		typedef boost::signals2::signal< bool( const T& ) > signal_t;
 
 		error_sig error_signal;
 		signal_t signal;
@@ -67,7 +67,7 @@ namespace Xmms
 	template<>
 	struct Signal< void > : public SignalInterface
 	{
-		typedef boost::signal< bool() > signal_t;
+		typedef boost::signals2::signal< bool() > signal_t;
 
 		error_sig error_signal;
 		signal_t signal;
@@ -212,7 +212,7 @@ namespace Xmms
 	inline bool
 	callSignal( const Signal< void >* sig, xmmsv_t*& /* val */)
 	{
-		return sig->signal();
+		return *sig->signal();
 	}
 
 	/** Called on the notifier udata when an xmmsc_result_t is freed.
@@ -244,7 +244,7 @@ namespace Xmms
 
 			std::string error( buf );
 			if( !data->error_signal.empty() ) {
-				ret = data->error_signal( error );
+				ret = *data->error_signal( error );
 			}
 
 		}
@@ -257,7 +257,7 @@ namespace Xmms
 				catch( std::exception& e ) {
 
 					if( !data->error_signal.empty() ) {
-						ret = data->error_signal( e.what() );
+						ret = *data->error_signal( e.what() );
 					}
 
 				}
@@ -272,7 +272,7 @@ namespace Xmms
 		return ret;
 	}
 
-	typedef boost::signal< void() > DisconnectCallback;
+	typedef boost::signals2::signal< void() > DisconnectCallback;
 
 	void disconnect_callback( void* userdata );
 	/** @endcond INTERNAL */
--- xmms2-0.6DrMattDestruction/src/include/xmmsclient/xmmsclient++/helpers.h.orig	2009-04-21 19:51:11.000000000 +0200
+++ xmms2-0.6DrMattDestruction/src/include/xmmsclient/xmmsclient++/helpers.h	2023-06-22 18:21:32.241241299 +0200
@@ -26,7 +26,7 @@
 
 #include <boost/bind.hpp>
 #include <boost/function.hpp>
-#include <boost/signal.hpp>
+#include <boost/signals2.hpp>
 #include <boost/type_traits/remove_pointer.hpp>
 
 #include <string>
