--- xmms2-0.1DR2.2/src/xmms/Program.orig	2023-05-08 21:47:48.436132214 +0200
+++ xmms2-0.1DR2.2/src/xmms/Program	2023-05-08 22:21:59.581686871 +0200
@@ -33,17 +33,6 @@ def config(env):
 	env.pkgconfig("gmodule-2.0", fail=True)
 	env.pkgconfig("gthread-2.0", fail=True)
 	env.pkgconfig("sqlite3", fail=True)
-	print "checking for sqlite version < 3.2.4...",
-	e = env.conf.TryRun("#include <sqlite3.h>\n int main() { if (SQLITE_VERSION_NUMBER > 3002004) { return 1; } else { return 0; } }", ".c")
-	if e[0] == 1:
-		print "ok!"
-	else:
-		print "no!"
-		print "*"*78
-		print "There is a KNOWN error with this version of SQLite and XMMS2."
-		print "PLEASE downgrade SQLite or UPGRADE XMMS2 to latest devel branch"
-		print "*"*78
-		raise RuntimeError("Abort!")
 		
 	if env.platform != 'win32':
 		env.checklib("m", "sin", fail=True)
