--- xmms2-0.8DrO_o/wscript.orig	2023-06-24 11:34:31.911090412 +0200
+++ xmms2-0.8DrO_o/wscript	2023-06-24 11:35:23.207479182 +0200
@@ -293,7 +293,7 @@ def configure(conf):
     else:
         dirty = changed and "-dirty" or ""
         conf.msg("uncommited changed", changed and "yes" or "no")
-        conf.env.VERSION = "%s (git commit: %s%s)" % (BASEVERSION, nam, dirty)
+        conf.env.VERSION = BASEVERSION
 
     if conf.options.with_profiling:
         conf.env.with_profiling = True
