--- xmms2-0.2DrDolittle/src/clients/lib/ruby/Library.orig	2023-05-10 22:10:51.425835301 +0200
+++ xmms2-0.2DrDolittle/src/clients/lib/ruby/Library	2023-05-10 22:34:00.468310213 +0200
@@ -17,16 +17,16 @@ def config(env):
 
 		print(version)
 
-		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(Config::CONFIG[\"archdir\"] + \"/ruby.h\")'").strip()
+		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(RbConfig::CONFIG[\"rubyhdrdir\"] + \"/ruby.h\")'").strip()
 		if got_ruby_h != "true" :
 			raise ConfigError("bad")
 
-		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(Config::CONFIG[\"archdir\"], 0..1)'"
+		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(RbConfig::CONFIG[\"rubyhdrdir\"], 0..1)'"
 		env.configcmd(cmd)
 		if "RUBYARCHDIR" in env:
 			env.librarypath = env["RUBYARCHDIR"]
 		else:
-			cmd = "ruby -rrbconfig -e 'print Config::CONFIG[\"sitearchdir\"]'"
+			cmd = "ruby -rrbconfig -e 'print RbConfig::CONFIG[\"vendorarchdir\"]'"
 			env.librarypath = os.popen(cmd).read().strip()
 
 		env.Prepend(LIBPATH="src/clients/lib/xmmsclient")
--- xmms2-0.2DrDolittle/src/clients/lib/ruby/Library.ecore.orig	2023-05-10 22:10:51.429168616 +0200
+++ xmms2-0.2DrDolittle/src/clients/lib/ruby/Library.ecore	2023-05-10 22:34:47.804720436 +0200
@@ -23,16 +23,16 @@ def config(env):
 
 		print(version)
 
-		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(Config::CONFIG[\"archdir\"] + \"/ruby.h\")'").strip()
+		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(RbConfig::CONFIG[\"rubyhdrdir\"] + \"/ruby.h\")'").strip()
 		if got_ruby_h != "true" :
 			raise ConfigError("")
 
-		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(Config::CONFIG[\"archdir\"], 0..1)'"
+		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(RbConfig::CONFIG[\"rubyhdrdir\"], 0..1)'"
 		env.configcmd(cmd)
 		if "RUBYARCHDIR" in env:
 			env.librarypath = env["RUBYARCHDIR"]
 		else:
-			cmd = "ruby -rrbconfig -e 'print Config::CONFIG[\"sitearchdir\"]'"
+			cmd = "ruby -rrbconfig -e 'print RbConfig::CONFIG[\"vendorarchdir\"]'"
 			env.librarypath = os.popen(cmd).read().strip()
 
 		env.Prepend(LIBPATH="src/clients/lib/xmmsclient")
--- xmms2-0.2DrDolittle/src/clients/lib/ruby/Library.glib.orig	2023-05-10 22:10:51.429168616 +0200
+++ xmms2-0.2DrDolittle/src/clients/lib/ruby/Library.glib	2023-05-10 22:35:11.117927471 +0200
@@ -17,16 +17,16 @@ def config(env):
 
 		print(version)
 
-		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(Config::CONFIG[\"archdir\"] + \"/ruby.h\")'").strip()
+		got_ruby_h = env.run("ruby -rrbconfig -e 'puts File.exist?(RbConfig::CONFIG[\"rubyhdrdir\"] + \"/ruby.h\")'").strip()
 		if got_ruby_h != "true" :
 			raise ConfigError("bad")
 
-		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(Config::CONFIG[\"archdir\"], 0..1)'"
+		cmd = "ruby -rrbconfig -e 'puts \"-I%s -L%s\" % [].fill(RbConfig::CONFIG[\"rubyhdrdir\"], 0..1)'"
 		env.configcmd(cmd)
 		if "RUBYARCHDIR" in env:
 			env.librarypath = env["RUBYARCHDIR"]
 		else:
-			cmd = "ruby -rrbconfig -e 'print Config::CONFIG[\"sitearchdir\"]'"
+			cmd = "ruby -rrbconfig -e 'print RbConfig::CONFIG[\"vendorarchdir\"]'"
 			env.librarypath = os.popen(cmd).read().strip()
 
 		env.pkgconfig("glib-2.0")
--- xmms2-0.2DrDolittle/src/clients/lib/ruby/rb_xmmsclient.c.orig	2023-05-10 22:10:51.432501932 +0200
+++ xmms2-0.2DrDolittle/src/clients/lib/ruby/rb_xmmsclient.c	2023-05-10 22:36:02.564315428 +0200
@@ -721,7 +721,7 @@ static VALUE c_playlist_add (VALUE self,
 
 	if (!NIL_P (rb_check_string_type (arg)))
 		is_str = true;
-	else if (rb_obj_is_kind_of (arg, rb_cFixnum))
+	else if (rb_obj_is_kind_of (arg, rb_cInteger))
 		is_str = false;
 	else {
 		rb_raise (eClientError, "unsupported argument");
@@ -755,7 +755,7 @@ static VALUE c_playlist_insert (VALUE se
 
 	if (!NIL_P (rb_check_string_type (arg)))
 		is_str = true;
-	else if (rb_obj_is_kind_of (arg, rb_cFixnum))
+	else if (rb_obj_is_kind_of (arg, rb_cInteger))
 		is_str = false;
 	else {
 		rb_raise (eClientError, "unsupported argument");
