--- xmms2-0.7DrNo/wscript.orig	2023-06-23 06:27:29.055856784 +0200
+++ xmms2-0.7DrNo/wscript	2023-06-23 06:28:26.602468011 +0200
@@ -275,7 +275,6 @@ def configure(conf):
         conf.check_message("uncommitted changes", "", bool(changed))
         conf.env["VERSION"] = BASEVERSION + " (git commit: %s%s)" % (nam, dirty)
 
-    conf.env["CCFLAGS"] = Utils.to_list(conf.env["CCFLAGS"]) + ['-g', '-O0']
     for warning in ('all',
                     'no-format-extra-args',
                     'no-format-zero-length',
@@ -295,7 +294,6 @@ def configure(conf):
             # autogenerate uselib definitions to disable warnings
             conf.env["CCFLAGS_NO%s" % warning.replace("-","").upper()] = ["-Wno-%s" % warning]
 
-    conf.env["CXXFLAGS"] = Utils.to_list(conf.env["CXXFLAGS"]) + ['-g', '-O0']
     conf.env['XMMS_PKGCONF_FILES'] = []
     conf.env['XMMS_OUTPUT_PLUGINS'] = [(-1, "NONE")]
 
