--- xmms2-0.2DrEvil/src/clients/lib/java/Library.orig	2023-05-11 22:00:10.991235882 +0200
+++ xmms2-0.2DrEvil/src/clients/lib/java/Library	2023-05-11 22:01:06.790933589 +0200
@@ -123,7 +123,7 @@ def config(env):
 	
 	env.Java(target = mydir, source = [mydir + '/src'])
 	env.Jar(target=mydir + '/xmms2java.jar', source=mydir + '/org')
-	env.Install(env['PREFIX'] + '/share/xmms2/java', mydir + '/xmms2java.jar')
+	env.Install(env['PREFIX'] + '/share/java', mydir + '/xmms2java.jar')
 	def mkdir(target, source, env):
 		try:
 			os.mkdir(mydir + "/src/org/xmms2/xmms2bindings")
